### PR TITLE
Fix BottomSheetBackdrop "falsey" defaults

### PR DESCRIPTION
**Pull request by thecodedrift**
This is a change to how `BottomSheetBackdrop` handles its falsey defaults. Values such as `0` and `false` will automatically assume their fallback values instead of allowing their literal `0` or `false` to be applied. There is a reproducible Snak in #779 which shows the problem.

https://snack.expo.dev/@jakobo/bottomsheetbackdrop---appearsonindex-bug

## Motivation

The fix is to ensure that the provided value was truly undefined before assigning the default. This patch uses nullish coalesce `??` to achieve that. The following code will never render a backdrop.

```tsx
const renderBackdrop = useCallback(
  props => (
    <BottomSheetBackdrop
      {...props}
      disappearsOnIndex={-1}
      appearsOnIndex={0}
    />
  ),
  []
);
```

Under the hood, `BottomSheetBackdrop` will assign `1` for `appearsOnIndex` because JS treats `0` as false-like.

## Contributing Guidelines ✅ 
* [x] Prefer small pull requests focused on one change.
* [x] Verify that linters and tests are passing.
  * **note** There were additional unrelated prettier errors which were fixed in the second commit
* [x] Review the documentation to make sure it looks good.
* [x] Follow the pull request template when opening a pull request.
* [x] For pull requests that change the API or implementation, discuss with maintainers first by opening an issue.

### CODE REVIEW

**Comment by thecodedrift** [comment]
@gorhom, let me know if there's anything I can do to help get this PR approved. I'll continue to rebase as necessary since the changes are scoped to mostly the Backdrop component

**Comment by jakobo** [comment]
!stale

**Comment by thecodedrift** [comment]
Rebase verification (and un-stale): `Current branch jakobo/779/BottomSheetBackdrop_defaults is up to date.`

**Comment by IngyuTae** [comment]
Hi, @gorhom Do you have any plan on this PR?

**Comment by jakobo** [comment]
Not stale. Still relevant. Still rebased. Still passing.

**Comment by thecodedrift** [comment]
@gorhom (ping) The fix is 3 Stalebots old. Would love to land this.

File: src/components/bottomSheetBackdrop/BottomSheetBackdrop.tsx [hunk 1]
```diff
@@ -24,10 +24,10 @@ import type { BottomSheetDefaultBackdropProps } from './types';
 
 const BottomSheetBackdropComponent = ({
   animatedIndex,
-  opacity = DEFAULT_OPACITY,
-  appearsOnIndex = DEFAULT_APPEARS_ON_INDEX,
-  disappearsOnIndex = DEFAULT_DISAPPEARS_ON_INDEX,
-  enableTouchThrough = DEFAULT_ENABLE_TOUCH_THROUGH,
+  opacity: userOpacity,
```
Comment (gorhom) [inline src/components/bottomSheetBackdrop/BottomSheetBackdrop.tsx]: could you follow the name convention, it should be `_providedOpacity`.

https://github.com/gorhom/react-native-bottom-sheet/blob/955b7748932ba5ea81d2406c0acf7b612fecbf0e/src/components/bottomSheet/BottomSheet.tsx#L97

**Review by gorhom — CHANGES_REQUESTED** [review]

**Comment by gorhom** [comment]
hi @jakobo, thanks for submitting this PR, i left one comment regarding the name convention for the props.

**Comment by thecodedrift** [comment]
Applied the requested changes, @gorhom. Let me know if there's any other changes. 🎉

**Review by gorhom — APPROVED** [review]
thanks @jakobo for updating the naming! this will be release on the next patch release

**Pull request merged**